Footer improvements and fixes

coverbeck opened the pull request:
dockstore/dockstore#3509

* Update copyright year by making it dynamic
* Change link to ga4gh/dockstore to dockstore/dockstore

dockstore/dockstore#3698

* Link for UI (and web service) works for actual commits. Try to figure out if the link should be to a tag
or commit and generate link accordingly.
* For web service, if the version is a SNAPSHOT jar (which it is on dev), or a `development-build`, show commit id

SEAB-1552

Add a "Build Info" link, which copies build info in markdown form to clipboard:

```
[Webservice](https://github.com/dockstore/dockstore/commits/981edd1) - 981edd1

[UI](https://github.com/dockstore/dockstore-ui2/commits/597aeeed) - 2.6.1-39-g597aeeed

[Compose Setup](https://github.com/dockstore/compose_setup/releases/tag/1.9.0) - 1.9.0

[Deploy](https://github.com/dockstore/dockstore-deploy/commits/3723b1a) - 3723b1a
```

which look like this:

[Webservice](https://github.com/dockstore/dockstore/commits/981edd1) - 981edd1

[UI](https://github.com/dockstore/dockstore-ui2/commits/597aeeed) - 2.6.1-39-g597aeeed

[Compose Setup](https://github.com/dockstore/compose_setup/releases/tag/1.9.0) - 1.9.0

[Deploy](https://github.com/dockstore/dockstore-deploy/commits/3723b1a) - 3723b1a

![Screen Shot 2020-09-08 at 9 20 14 AM](https://user-images.githubusercontent.com/1049340/92502619-0399b100-f1b5-11ea-9c76-6d8dda519d2e.png)

garyluu commented on src/app/footer/footer.component.ts:
```diff
@@ -59,7 +61,12 @@ export class FooterComponent extends Base implements OnInit {
           if (metadata.hasOwnProperty('version')) {
-            this.version = metadata['version'];
+            const metadatum = metadata['version'];
+            if ((metadatum && metadatum.indexOf('SNAPSHOT') !== -1) || metadatum.indexOf('development-build') !== -1) {
```
i feel like there's another alternative to indexOf !== -1

coverbeck replied:
Hmm, did not know. There is `includes`, which is not supported on IE, but I guess target es5 should address that. We use it in other places, so updated it.

coverbeck commented:
> With four links, maybe consider a copy to clipboard option for git issue reporting but also to collapse Deploy / Infrastructure
> https://efficientuser.com/2019/04/04/angular-copy-to-clipboard-feature/
> 
> Right now, I copy the whole thing into an issue and then strip out the icon. But the "Deploy / Infrastructure" won't be a link if I do that, so maybe just forget human-readable for that line and add an icon for copy-paste.

Do you want one copy button that copies all 4 commits/releases (webservice, UI, deploy, and infra)? Or 4 copy buttons? And what do you want it to copy -- the fully qualified urls, or only the commit ids/tags?

garyluu commented on package.json:
```diff
@@ -3,7 +3,7 @@
   "license": "Apache License 2.0",
   "config": {
-    "webservice_version": "1.10.0-alpha.0"
+    "webservice_version": "https://5092-33383826-gh.circle-artifacts.com/0/tmp/artifacts/dockstore-webservice-1.10.0-alpha.1-SNAPSHOT.jar"
```
should be something like https://app.circleci.com/pipelines/github/dockstore/dockstore/4188/workflows/9c62995e-d9e5-4b23-bf51-37c6bdc8c12f/jobs/4674/artifacts which gives more information

garyluu replied:
.

coverbeck replied:
Rebased off develop with Natalie's changes and I'm just using that.

garyluu commented on src/app/footer/footer.component.ts:
```diff
@@ -59,7 +63,19 @@ export class FooterComponent extends Base implements OnInit {
+              Dockstore.COMPOSE_SETUP_VERSION,
+              Dockstore.DEPLOY_VERSION
+            );
+            console.log('content', this.content);
```
leftover?

coverbeck replied:
Yep, thanks fixed.

garyluu commented:
One other thing, why is the clipboard button actually a link?

coverbeck commented:
> One other thing, why is the clipboard button actually a link?

Seemed consistent with the rest of the footer, but I don't feel strongly about it. I tried the whole thing as a button briefly, and it didn't look too good, although I didn't spend a lot of time trying to style it.

Are you suggesting it be a button instead? And if so, the icon + text, or just the icon, and not have the text be clickable?

garyluu commented:
> > One other thing, why is the clipboard button actually a link?
> 
> Seemed consistent with the rest of the footer, but I don't feel strongly about it. I tried the whole thing as a button briefly, and it didn't look too good, although I didn't spend a lot of time trying to style it.
> 
> Are you suggesting it be a button instead? And if so, the icon + text, or just the icon, and not have the text be clickable?



> > One other thing, why is the clipboard button actually a link?
> 
> Seemed consistent with the rest of the footer, but I don't feel strongly about it. I tried the whole thing as a button briefly, and it didn't look too good, although I didn't spend a lot of time trying to style it.
> 
> Are you suggesting it be a button instead? And if so, the icon + text, or just the icon, and not have the text be clickable?

Just trying to avoid https://stackoverflow.com/a/1293130  . The rest of the footer are links because...they have links.

coverbeck commented:
Changed to button, moved to bottom of column.

![Screen Shot 2020-09-08 at 6 02 27 PM](https://user-images.githubusercontent.com/1049340/92542573-f48b2100-f1fd-11ea-9d86-b5fe3abfc43d.png)
